builtins/amp-img.js

```diff
@@ -47,13 +47,20 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
-  attributeChangedCallback(name, unusedOldValue, unusedNewValue) {
-    if (name === 'src') {
+  mutatedAttributesCallback(mutations) {
+    if (mutations['src'] !== undefined || mutations['srcset'] !== undefined) {
       this.srcset_ = srcsetFromElement(this.element);
-      this.updateImageSrc_();
-    } else if (this.img_ && ATTRIBUTES_TO_PROPAGATE.indexOf(name) >= 0) {
-      this.propagateAttributes(name, this.img_,
-          /* opt_removeMissingAttrs */ true);
+      // This element may not have been laid out yet.
+      if (this.img_) {
+        this.updateImageSrc_();
+      }
+    }
+
+    if (this.img_) {
```

Comment:
We always want to update `this.srcset_` though, how would that work without duplicating the first conditional?